sql/src/main/java/io/crate/planner/SubqueryPlanner.java

```diff
@@ -47,8 +46,8 @@ private void planSubquery(SelectSymbol selectSymbol) {
         subQueries.put(subPlan, selectSymbol);
     }
 
-    public Map<Plan, SelectSymbol> planSubQueries(QuerySpec querySpec) {
-        querySpec.visitSymbols(visitor);
+    public Map<Plan, SelectSymbol> planSubQueries(Consumer<Consumer<? super Symbol>> symbolVisitor) {
```

Comment:
Maybe just a personal note but at least for me, this Consumer<Consumer ... > type bears a high cognitive load (especially as the visitor reverses the logic as well) 
Can we maybe extract `visitSymbols` from `QueriedRelation` into its own interface?

Reply:
Changed it to take the `QueriedRelation` instead for now